### PR TITLE
{test} Add subscription to resource group preparer

### PR DESCRIPTION
**Pull request by Juliehzl**
**Description**

Add `subscription` to allow users to customize subscription for resource group

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli-testsdk/azure/cli/testsdk/preparers.py [hunk 1]
```diff
@@ -77,6 +78,8 @@ def create_resource(self, name, **kwargs):
             tags['job'] = os.environ['ENV_JOB_NAME']
         tags = ' '.join(['{}={}'.format(key, value) for key, value in tags.items()])
         template = 'az group create --location {} --name {} --tag ' + tags
+        if self.subscription:
+            template += '--subscription '.format(self.subscription)
```
Comment (kairu-ms) [inline src/azure-cli-testsdk/azure/cli/testsdk/preparers.py:82]: I think you also need to add it in remove_resource function

**Comment by yonzhan** [comment]
test

File: src/azure-cli-testsdk/azure/cli/testsdk/preparers.py [hunk 2]
```diff
@@ -85,7 +88,10 @@ def create_resource(self, name, **kwargs):
     def remove_resource(self, name, **kwargs):
         # delete group if test is being recorded and if the group is not a dev rg
         if not self.dev_setting_name:
-            self.live_only_execute(self.cli_ctx, 'az group delete --name {} --yes --no-wait'.format(name))
+            template = 'az group delete --name {} --yes --no-wait '
+            if self.subscription:
+                template += '--subscription '.format(self.subscription)
```
Comment (kairu-ms) [inline src/azure-cli-testsdk/azure/cli/testsdk/preparers.py:93]:
```suggestion
                template += '--subscription {}'.format(self.subscription)
```

Comment (Juliehzl) [reply src/azure-cli-testsdk/azure/cli/testsdk/preparers.py]: updated in new commit

**Review by kairu-ms — APPROVED** [review]

**Review by jsntcy — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
There was a error with the local test:
> azure.core.exceptions.ResourceNotFoundError: (ResourceGroupNotFound) Resource group 'cli_test_vm_cross_tenant_s6q563i27wt562ac7xsoxnxrlw5kqokvps2quwdxju33xme6l2' could not be found.

**Pull request merged**